sdk/keyvault/keyvault-admin/src/lro/restore/operation.ts

```diff
@@ -0,0 +1,271 @@
+  /**
+   * Options for the core-http requests.
+   */
+  requestOptions?: RequestOptionsBase;
+  /**
+   * An interface representing the internal KeyVaultClient.
+   */
+  client?: KeyVaultClient;
+  /**
+   * The base URL to the vault
+   */
+  vaultUrl: string;
+  /**
+   * The URI of the blob storage account.
+   */
+  blobStorageUri: string;
+  /**
+   * The SAS token.
+   */
+  sasToken: string;
+  /**
+   * The Folder name of the blob where the previous successful full backup was stored
+   */
+  folderName: string;
+  /**
+   * The id returned as part of the restore request
+   */
+  jobId?: string;
+  /**
+   * Status of the restore operation.
+   */
+  status?: string;
+  /**
+   * The status details of restore operation.
+   */
+  statusDetails?: string;
+  /**
+   * The start time of the restore operation in UTC
+   */
+  startTime?: Date;
+  /**
+   * The end time of the restore operation in UTC
+   */
+  endTime?: Date;
+}
+
+/**
+ * An interface representing a restore Key Vault's poll operation.
+ */
+export interface RestorePollOperation extends PollOperation<RestorePollOperationState, string> {}
+
+/**
+ * Tracing the fullRestore operation
+ */
+async function fullRestore(
+  client: KeyVaultClient,
+  vaultUrl: string,
+  options: KeyVaultClientFullRestoreOperationOptionalParams
+): Promise<KeyVaultClientFullRestoreOperationResponse> {
+  const span = createSpan("generatedClient.fullRestore", options);
+
+  let response: KeyVaultClientFullRestoreOperationResponse;
+  try {
+    response = await client.fullRestoreOperation(vaultUrl, setParentSpan(span, options));
+  } finally {
+    span.end();
+  }
+
+  return response;
+}
+
+/**
+ * Tracing the fullRestoreStatus operation.
+ */
+async function fullBackupStatus(
+  client: KeyVaultClient,
+  vaultUrl: string,
+  jobId: string,
+  options: OperationOptions
+): Promise<KeyVaultClientFullBackupStatusResponse> {
+  const span = createSpan("generatedClient.fullBackupStatus", options);
+
+  let response: KeyVaultClientFullBackupStatusResponse;
+  try {
+    response = await client.fullBackupStatus(vaultUrl, jobId, setParentSpan(span, options));
+  } finally {
+    span.end();
+  }
+
+  return response;
+}
+
+/**
+ * @summary Reaches to the service and updates the restore poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: RestorePollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: RestorePollOperationState) => void;
+  } = {}
+): Promise<RestorePollOperation> {
+  const state = this.state;
+  const { vaultUrl, blobStorageUri, sasToken, folderName } = state;
+
+  // Internal properties,
+  // the reference is only potentially undefined in the public representation of the poller.
+  const client = state.client!;
```

Comment:
I think for me if you use `!` on a shape you control it feels like a code smell, since either it shouldn't be optional or you should check it... otherwise what's the point of TS checking it for you?

Reply:
Fixed! Please review again.

Reply:
Thank you for your feedback btw!